framework/include/araui/ui_widget.h

```diff
@@ -0,0 +1,722 @@
+ *
+ * @param[in] widget Handle of the widget
+ * @param[in] degree Degree of the rotated widget (-360 ~ 360 degree)
+ * @return On success, UI_OK is returned. On failure, the defined error type is returned.
+ *
+ * @see ui_error_t
+ */
+ui_error_t ui_widget_set_rotation(ui_widget_t widget, int32_t degree);
+
+/**
+ * @brief Set the userdata of the widget.
+ *
+ * Each widget can have its own data (void * type).
+ *
+ * @param[in] widget Handle of the widget
+ * @param[in] userdata Userdata pointer
+ * @return On success, UI_OK is returned. On failure, the defined error type is returned.
+ *
+ * @see ui_widget_get_userdata
+ */
+ui_error_t ui_widget_set_userdata(ui_widget_t widget, void *userdata);
+
+/**
+ * @brief Get the userdata of the widget.
+ *
+ * Each widget can have its own data (void * type).
+ *
+ * @param[in] widget Handle of the widget
+ * @return On success, userdata pointer is returned. On failure, Null value is returned.
+ *
+ * @see ui_widget_set_userdata
+ */
+void *ui_widget_get_userdata(ui_widget_t widget);
+
+
+/********************************
+ *
+ * Image widget
+ *
+ */
+
+/**
+ * @brief Create a image widget.
+ *
+ * In order to operate properly, the created widget needs to set a parent (@ref ui_widget_add_child).
+ *
+ * The input of this API (image) can be NULL value.
+ * If user want to set the image later, please use ui_image_widget_set_image() (@ref ui_image_widget_set_image).
+ *
+ * Width and height are not specified when registering an image widget.
+ * Width and height of the image widget are automatically calculated from the specified image asset.
+ * If the width and height of image widget are different from the image asset, please resize the image widget by using @ref ui_widget_set_size.
+ *
+ * @param[in] image Handle of the image asset
+ * @return On success, the handle of image widget is returned. On failure, UI_NULL is returned.
+ *
+ * @see ui_image_asset_create_from_file()
+ * @see ui_widget_add_child()
+ * @see ui_asset_t
+ */
+ui_widget_t ui_image_widget_create(ui_asset_t image);
+
+/**
+ * @brief Set the image asset to be used by the image widget.
+ *
+ * In order to prevent the memory leakage problem, the image assets which will not be used should be destroyed.
+ * Please refer @ref ui_image_asset_destroy().
+ *
+ * @param[in] widget Handle of the image widget
+ * @param[in] image Handle of the new image asset
+ * @return On success, UI_OK is returned. On failure, the defined error type is returned.
+ *
+ * @see ui_image_asset_create_from_buffer()
+ * @see ui_image_asset_destroy()
+ */
+ui_error_t ui_image_widget_set_image(ui_widget_t widget, ui_asset_t image);
+
+/**
+ * @brief Set the crop area of the image asset in the image widget.
+ *
+ * Crop the image asset which set to the image widget.
+ * If the image asset of image widget is not set, UI_INVALID_PARAM is retured.
+ * Furthermore, a crop area can not exceed the original image asset.
+ * If user calls ui_image_widget_set_crop_area(),
```

Comment:
As `ui_image_widget_resize_image_asset()` is removed, please remove descriptions about this API.